PORTING.md

```diff
@@ -0,0 +1,47 @@
+Porting applications to use zlib-ng
+-----------------------------------
+
+Zlib-ng can be used/compiled in two different modes, that require some
+consideration by the application developer.
+
+zlib-compat mode
+----------------
+Zlib-ng can be compiled in zlib-compat mode, suitable for zlib-replacement
+in a single application or system-wide.
+
+Please note that zlib-ng in zlib-compat mode is API-compatible but not
+ABI-compatible, meaning that you cannot simply replace the zlib library/dll
+files and expect the application to work. The application will need to be
+recompiled against the zlib-ng headers and libs to ensure full compatability.
+
+Compile against the zlib.h provided by zlib-ng.
+
+Advantages:
+- Easy to port to, since it only requires a recompile of the application and
+  no changes to the application code.
+
+Disadvantages:
+- Can conflict with a system-installed zlib, as that can often be linked in
```

Comment:
> Can conflict with system-installed zlib..

- _When installed it_ can conflict with system-installed zlib..?

Maybe the rest of it would need to be reworded too.

Reply:
If I understand correctly, it can still conflict if zlib-ng is statically linked in app/lib and zlib is also for some reason linked in by something else. As I understand, the exact interactions when two conflicting definitions are loaded into the same process, it results in an undefined condition where some functions can be selected from zlib-ng and some from zlib.

Luckily we implemented --fno-semantic-interposisiton, so at least that makes sure zlib-ng will never call zlib functions directly, but the application can still get confused and mix what it uses.

Reply:
Ok sounds good, I was thinking it would only conflict if installed like the README.md suggests.